docs/build/authentication.md

```diff
@@ -209,6 +209,18 @@ The authentication process varies depending on which wallet or provider you are
     const provider = new Ed25519Provider(seed)
     ```
 
+## Ensure you have a DID instance
+If you are using the [core client](../../reference/javascript/clients/#core-client) then you should
+already have a DID installed in your client from following the instructions in the [Configure your DID](configure-did.md) page. If you are using the [http client](../../reference/javascript/clients/#http-client), however, then you will need to set up a DID instance for your client now.
+
+``` javascript
+import { DID } from 'dids'
+
+if (ceramic.did == undefined) {
+  ceramic.did = new DID()
+}
```

Comment:
Can you add the `key-did-resolver` in this example? A dids instance without any resolvers won't help you much :)

Reply:
Why?  This is for the HTTP-client case, which never has to actually resolve DIDs since all the validation happens in the daemon.  The only reason we need the DID instance is to give it a DID Provider to use to perform writes.

Reply:
So why give the example with an empty one? That doesn't make sense to me. Then we should also include the `key-did-provider-ed25519` for completeness. Just including an empty DID instance is very incomplete. Either we remove it or include a provider.

Reply:
The provider is set in the next section.  This whole page is about how to set a provider so you can perform writes.  This section with setting the empty DID is to make the overall flow work for both the core client, where you'll already have a DID with Resolvers loaded, and for the http client, where you likely won't have a DID yet.

Reply:
Yeah, as the flow is right now that won't even work since there is no did-resolver.
I don't really think it makes sense to separate this into different sections. Makes more sense to include as one code snippet rather than multiple. I don't think it ever makes sense to pass an empty DID instance.

Reply:
Okay, updated to combine the snippet for setting the provider with the one for ensuring that there is a DID instance

Reply:
> Yeah, as the flow is right now that won't even work since there is no did-resolver.

I actually want to confirm something here - do you actually need a Resolver in your DID instance if you're not resolving dids?  Since the HTTP client doesn't do any validation of signatures, my assumption was that the HTTP client doesn't need a DID Resolver. It needs a DID Provider to author commits, but if it's being used only to load streams then the HTTP client shouldn't even need a DID instance at all, right?

Reply:
> We need a did-resolver when calling `authenticate()` since this call includes a signature from the DID which is being authenticated. Only way to verify this signature is to have the resolver.

So it sounds then like the times when you need a DID Resolver are the exact same times when you need a DID Provider.  So maybe the whole `Configure your DID` and `Authentication` pages should be combined into one flow for setting both the Resolver and the Provider?  Is there ever a use case for a `core` or `http-client` Ceramic client to have a DID with a Resolver but not a Provider?